src/de/espend/idea/laravel/blade/actions/ExtractPartialViewHandler.java

```diff
@@ -0,0 +1,127 @@
+
+        if(StringUtils.isBlank(selectedText)) {
+            return;
+        }
+
+        PsiDirectory targetDirectory = getViewsDirectory(project, psiFile);
+
+        if(targetDirectory == null) {
+            return;
+        }
+
+        ExtractPartialViewDialog dialog = new ExtractPartialViewDialog(project, targetDirectory.getVirtualFile());
+        dialog.show();
+
+        if(!dialog.isOK()) {
+            return;
+        }
+
+        final String viewName = dialog.getViewName();
+        final String canonizedViewName = viewName.replace('/', '.');
+
+        ApplicationManager.getApplication().runWriteAction(() -> {
+            CommandProcessor.getInstance().executeCommand(editor.getProject(), () -> {
+
+                final String[] viewPathParts = canonizedViewName.split("\\.");
+
+                PsiDirectory directory = targetDirectory;
+
+                for(int i = 0; i < viewPathParts.length - 1; i++) {
+                    PsiDirectory newDirectory = directory.findSubdirectory(viewPathParts[i]);
+
+                    if(newDirectory == null) {
+                        newDirectory = directory.createSubdirectory(viewPathParts[i]);
+                    }
+
+                    directory = newDirectory;
+                }
+
+                final String fileName = viewPathParts[viewPathParts.length - 1] + ".blade.php";
+
+                if(directory.findFile(fileName) != null) {
+                    Messages.showMessageDialog(project, "File already exists", "Info", Messages.getErrorIcon());
+                    return;
+                }
+
+                directory.add(PsiFileFactory.getInstance(project).createFileFromText(fileName,
+                        BladeLanguage.INSTANCE,
+                        selectedText));
+
+                int selectionStart = editor.getSelectionModel().getSelectionStart();
+                int selectionEnd = editor.getSelectionModel().getSelectionEnd();
+                editor.getDocument().replaceString(selectionStart,
+                        selectionEnd,
+                        "@include('" + viewName + "')");
+
+                editor.getSelectionModel().removeSelection();
+                editor.getCaretModel().moveToOffset(selectionStart);
+
+            }, "Extract partial view", editor.getDocument());
+        });
+    }
+
+    @Override
+    public void invoke(@NotNull Project project, @NotNull PsiElement[] psiElements, DataContext dataContext) {
+        // Will work only on editor, so this method won't be called
+    }
+
+    @Nullable
+    private PsiDirectory getViewsDirectory(Project project, PsiFile psiFile)
+    {
+        String basePath = project.getBasePath();
+
+        PsiDirectory directory = psiFile.getContainingDirectory();
+        while(directory != null && !directory.getVirtualFile().getPath().equals(basePath)) {
+            if(directory.getName().equals("views") || directory.getName().equals("templates")) {
```

Comment:
maybe iterate through  `ViewCollector.getPaths` and try to check if directory is inside this path. so we can then support custom project structures as well and do not really on `views` or `template` structure